Trappist/src/Promact.Trappist.Core/Controllers/TestSettingsController.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Promact.Trappist.DomainModel.Models.Test;
+using Promact.Trappist.Repository.TestSettings;
+
+namespace Promact.Trappist.Core.Controllers
+{
+    public class TestSettingsController : Controller
+    {
+        private readonly ITestSettingsRepository _settingsRepository;
+
+        public TestSettingsController(ITestSettingsRepository settingsRepository)
+        {
+            _settingsRepository = settingsRepository;
+        }
+
+        /// <summary>
+        /// Get the Settings saved for a particular Test
+        /// </summary>
+        /// <returns>Settings set for that Test</returns>
+        [HttpGet]
+        public IActionResult GetTestSettings()
+        {
+            var settings = _settingsRepository.GetTestSettings();
+            return Json(settings);
+        }
+
+        /// <summary>
+        /// Updates the changes made to the Settings of a Test
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="testObject"></param>
+        /// <returns>Updated Settings of that Test</returns>
+        [HttpPut("api/put/{id}")]
+        public IActionResult UpdateTestSettings([FromRoute] int id, [FromBody] Test testObject)
```

Comment:
In testObject, din't you get the id of the object? then why you have usedparameter "id" seprately?